python/cudf_polars/cudf_polars/dsl/expressions/rolling.py

```diff
@@ -210,9 +260,341 @@ def __init__(
             for ne in self.named_aggs
             for v in (ne.value,)
             if isinstance(v, expr.Agg)
+            or (
+                isinstance(v, expr.UnaryFunction)
+                and v.name in {"rank", "fill_null_with_strategy", "cum_sum"}
+            )
         ]
         self.by_count = len(by_expr)
-        self.children = tuple(by_expr) + tuple(child_deps)
+        self.children = tuple(by_expr) + tuple(self._order_by_exprs) + tuple(child_deps)
```

Comment:
```suggestion
        self.children = tuple(itertools.chain(by_expr, self._order_by_exprs, child_deps))
```
(To avoid the intermediate Python objects)